### ISSUE TITLE
False-positive `PLW1641` when setting `__hash__ = <ParentClass>.__hash__`

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#10557]
with v0.3.4, `ruff check t.py --select PLW1641` on

```python
class A:
    def __eq__(self, other):
        return False

    def __hash__(self):
        return 7

class B(A):
    def __eq__(self, other):
        return True

    __hash__ = A.__hash__
```

issues

```py
PLW1641 Object does not implement `__hash__` method
   |
12 | class B(A):
   |       ^ PLW1641
13 |     def __eq__(self, other):
14 |         return True
```

even though [the docs explicitly recommend this](https://docs.python.org/3/reference/datamodel.html#object.__hash__):

> If a class that overrides [__eq__()](https://docs.python.org/3/reference/datamodel.html#object.__eq__) needs to retain the implementation of [__hash__()](https://docs.python.org/3/reference/datamodel.html#object.__hash__) from a parent class, the interpreter must be told this explicitly by setting `__hash__ = <ParentClass>.__hash__`.

**Comment by charliermarsh** [comment]
This is now fixed.
